lib/Conversion/TorchToLinalg/DataMovement.cpp

```diff
@@ -1658,20 +1658,14 @@ class ConvertAtenSqueezeDimOp : public OpConversionPattern<AtenSqueezeDimOp> {
     if (!isValidDim(dim, inputRank))
       return rewriter.notifyMatchFailure(op, "dim is statically invalid");
 
-    // TODO: Handle the case where the dim(th) dimension is dynamic.
-    if (inputType.isDynamicDim(dim)) {
-      return rewriter.notifyMatchFailure(
-          op, "unimplemented: dim(th) dimension is not expected to be dynamic");
-    }
-
     const TypeConverter *typeConverter = getTypeConverter();
     auto resultType =
         cast<RankedTensorType>(typeConverter->convertType(op.getType()));
     int64_t resultRank = resultType.getRank();
 
     // If the dim(th) dimension of operand tensor type is not statically unit,
     // `aten.squeeze` will behave as an identity operation.
-    if (inputType.getDimSize(dim) != 1) {
+    if (inputType.getDimSize(dim) != 1 && !inputType.isDynamicDim(dim)) {
       rewriter.replaceOpWithNewOp<tensor::CastOp>(op, resultType, input);
       return success();
     }
```

Comment:
Since after this point we might be handling a dynamic dim that needs to be 1, it might be a good idea to include an assertion like
```cpp
if (inputType.isDynamicDim(dim)) {
   // assert dynamic squeeze dim size == 1
   Value cstDim = rewriter.create<arith::ConstantIndexOp>(op.getLoc(), dim);
   Value dimVal = rewriter.create<tensor::DimOp>(op.getLoc(), input, cstDim);
   Value cstOne = rewriter.create<arith::ConstantIndexOp>(op.getLoc(), 1);
   Value cmp = rewriter.create<arith::CmpIOp>(op.getLoc(), arith::CmpIPredicate::eq, dimVal, cstOne);
   rewriter.create<cf::AssertOp>(op.getLoc(), cmp, rewriter.getStringAttr("Expected dynamic squeeze dim size to be statically 1"));
}
```

This is definitely getting created already when lowering `PrimsSqueezeOp`, or more generally anywhere calling the util function `squeezeTensor`; however, it's probably better to have some redundancy in catching this kind of issue.

Reply:
There is already a check in `squeezeTensor`, but yeah it makes sense to add it.